### PR TITLE
Create a task from a message

### PR DESCRIPTION
**Pull request by hamza221**
closes [#6127](https://github.com/nextcloud/mail/issues/6127)

- [x] Move task.js and calendar.js to  [calendar-js](https://github.com/nextcloud/calendar-js) ?
- [x] ~~Wait for https://github.com/nextcloud/nextcloud-vue/pull/3945 to get merged and add the feature in Create event modal~~ follow-up
- [ ] Fix warns
image

### CODE REVIEW

**Comment by hamza221** [comment]
@ChristophWurst  lmk if you think task.js and calendar.js should be moved [calendar-js](https://github.com/nextcloud/calendar-js) when you find time to review it

File: src/store/calendar.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
+/**
+ * Maps a dav collection to our calendar object model
```
Comment (ChristophWurst) [inline src/store/calendar.js:2]: *our* model is the one used in Calendar, right? unsure if we want to push that into calendar-js. @st3iny what do you think?

Comment (ChristophWurst) [reply src/store/calendar.js:2]: we will leave it as-is

**Comment by ChristophWurst** [comment]
conflicts

**Review by ChristophWurst — APPROVED** [review]
Code looks good. Didn't test

**Comment by ChristophWurst** [comment]
eslint needs a fix

File: src/task.js [hunk 2]
```diff
@@ -0,0 +1,719 @@
...
+
+		this.syncStatus = null
+
+		// Time in seconds before the task is going to be deleted
+		this.deleteCountdown = null
+
+		// Queue for update requests with concurrency 1,
+		// because we only want to allow one request at a time
+		// (otherwise we will run into problems with changed ETags).
+		this.updateQueue = new PQueue({ concurrency: 1 })
```
Comment (ChristophWurst) [inline src/task.js:72]: never used -> drop pqueue

**Comment by JohannesGGE** [comment]
Screenshot from 2023-05-03 12-07-47_
I was a bit confused at first because I used the action-menu on the right and couldn't find the option. Maybe it should be added there too?

**Comment by JohannesGGE** [comment]
image
I noticed some things:
- The headline is "Create event", is that intended?
- I cannot select the calendar to which the task should be added (list is empty)
- Unlike the create event, it is not clear that the datetimes are "Start date" and "Due date" (at least for me) (most of the time I personally set only a due date)
- "Tell us your story:" is the description field? If yes, I would just write "Description" to prevent confusion.

**Comment by ChristophWurst** [comment]
*Create* should be disabled if you have no calendars with VTODO support, I guess

**Comment by hamza221** [comment]
> _Create_ should be disabled if you have no calendars with VTODO support, I guess

on it

**Comment by ChristophWurst** [comment]
@JohannesGGE please re-review :)

**Review by jancborchardt — COMMENTED** [review]
- [ ] The height of the description field could be increased so that it’s nicer to type in there, and that the "Create" button doesn’t float in the air but is exactly at the bottom.

File: src/components/TaskModal.vue [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+				</template>
+				<template #singleLabel="{option}">
+					<CalendarPickerOption
+						:display-icon="option.displayIcon"
+						v-bind="option" />
+				</template>
+			</Multiselect>
+
+			<br>
+			<button class="primary" :disabled="saving" @click="onSave">
```
Comment (JohannesGGE) [inline src/components/TaskModal.vue:57]:
```suggestion
			<button class="primary" :disabled="disabled" @click="onSave">
```

**Review by JohannesGGE — COMMENTED** [review]
I would format the date+time (all day disabled) like the "create event" does it. Right now there is just the time shown.
Screenshot from 2023-05-04 12-02-17
image

**Comment by ChristophWurst** [comment]
Nit: *Select option* -> *Select calendar*
Nit: *List is empty* -> *No calendars with task list support*

**Comment by hamza221** [comment]
image

**Review by ChristophWurst — APPROVED** [review]
Works

**Review by JohannesGGE — APPROVED** [review]

**Comment by ChristophWurst** [comment]
Please squash

**Pull request merged**